sync/src/synchronizer/block_fetcher.rs

```diff
@@ -123,26 +124,43 @@ impl BlockFetcher {
                     break;
                 }
 
-                let to_fetch = self
-                    .snapshot
-                    .get_ancestor(&best_known_header.hash(), index_height)?;
-
-                // NOTE: Filtering `BLOCK_STORED` but not `BLOCK_RECEIVED`, is for avoiding
-                // stopping synchronization even when orphan_pool maintains dirty items by bugs.
-                if self
-                    .snapshot
-                    .contains_block_status(&to_fetch.hash(), BlockStatus::BLOCK_STORED)
-                {
-                    continue;
+                // Try to get [from, to] headers, return the corresponding headers in reversed order.
```

Comment:
We could increase the stripe of each fetch as well, like bitcoin, which uses 128. The method `get_ancestor` may be as heavy as traversing 100 blocks, we should take the advantage of each `get_ancestor`.

However, the change requires more benchmarks, we could introduce it in a new PR.